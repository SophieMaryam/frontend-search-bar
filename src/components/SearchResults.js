import React from "react";
import "../index.css";

const SearchResults = ({ keyword, searchResults }) => {
  const showSearchResults = () => {
    // eslint-disable-next-line
    return searchResults.map((item, index) => {
      if (item.searchterm.includes(keyword.toLowerCase()) && keyword.length > 1) {
        return (
          <div className="text-bold" key={index}>
            <ul>
              <li className="input-fields search-results">
               {applyBoldStyle(item.searchterm)} ({item.nrResults})
              </li>
            </ul>
          </div>
        );
      }
    });
  };

  const applyBoldStyle = (input) => {
    var re = new RegExp(keyword, 'g')
    return (
        <div dangerouslySetInnerHTML={{
            __html: input.replace(re, '<span>' + keyword + '</span>')
        }} />
    )
  }
  
  return <div>{ keyword ? showSearchResults() : null }</div>;
};

export default SearchResults;