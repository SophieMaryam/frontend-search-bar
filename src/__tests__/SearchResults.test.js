import React from "react";
import { mount } from "enzyme";
import SearchResults from "../components/SearchResults";

let wrapper;
jest.fn("../apis/api")
describe("SearchResults component", () => {
  beforeEach(() => {
    wrapper = mount(<SearchResults />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("returns the default empty array when there is no data to map through", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("doesn't break without clothes data list", () => {
    expect(wrapper.find('li')).toHaveLength(0);                                                                                                                                           
  });

  it("doesn't break with an empty array", () => {
    wrapper = mount(<SearchResults keyword="" searchResults={[]} />);
    expect(wrapper.find('li')).toHaveLength(0);                                                                                                                                           
  });

  it("renders search results in li based on searchResults length", () => {
    const props = {
      keyword: "trui",
      searchResults: [{ searchterm: "kenzo trui", nrResults: 1000 }]
    }
    wrapper = mount(<SearchResults {...props} />);
    expect(wrapper.find("li")).toHaveLength(1);                                                                                                                                           
  });
});

