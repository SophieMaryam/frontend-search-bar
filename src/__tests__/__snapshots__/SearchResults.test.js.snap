// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component returns the default empty array when there is no data to map through 1`] = `ReactWrapper {}`;
